LeaderboardBackend.Test/Users.cs

```diff
@@ -0,0 +1,276 @@
+                services.AddSingleton<IClock, FakeClock>(_ => _clock)
+            )
+        );
+
+        _apiClient = new TestApiClient(_factory.CreateClient());
+        using IServiceScope scope = _factory.Services.CreateScope();
+        ApplicationContext context = scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+        await TestApiFactory.ResetDatabase(context);
+        _jwt = (await _apiClient.LoginAdminUser()).Token;
+    }
+
+    [OneTimeTearDown]
+    public void OneTimeTearDown() => _factory.Dispose();
+
+    [Test]
+    public async Task BanUser_OK()
+    {
+        IServiceScope scope = _factory.Services.CreateScope();
+        IUserService userService = scope.ServiceProvider.GetRequiredService<IUserService>();
+        ApplicationContext context = scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        RegisterRequest registerRequest = new()
+        {
+            Email = "testuser.banuser.ok@example.com",
+            Password = "Passw0rd",
+            Username = "BanUserTestOk"
+        };
+
+        CreateUserResult createUserResult = await userService.CreateUser(registerRequest);
+        createUserResult.IsT0.Should().BeTrue();
+        User user = createUserResult.AsT0;
+        context.ChangeTracker.Clear();
+
+        await _apiClient.Patch($"/users/{user.Id.ToUrlSafeBase64String()}", new()
+        {
+            Body = new
+            {
+                Role = UserRole.Banned,
+            },
+            Jwt = _jwt
+        });
+
+        User? res = await context.Users.FindAsync(user.Id);
+        res!.Role.Should().Be(UserRole.Banned);
+    }
+
+    [Test]
+    public async Task BanUser_Unauthenticated()
+    {
+        IServiceScope scope = _factory.Services.CreateScope();
+        IUserService userService = scope.ServiceProvider.GetRequiredService<IUserService>();
+        ApplicationContext context = scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        RegisterRequest registerRequest = new()
+        {
+            Email = "testuser.banuser.unauthn@example.com",
+            Password = "Passw0rd",
+            Username = "BanUserTestUnauthN"
+        };
+
+        CreateUserResult createUserResult = await userService.CreateUser(registerRequest);
+        createUserResult.IsT0.Should().BeTrue();
+        User user = createUserResult.AsT0;
+        context.ChangeTracker.Clear();
+
+        await _apiClient.Awaiting(a => a.Patch(
+            $"/users/{user.Id.ToUrlSafeBase64String()}",
+            new()
+            {
+                Body = new
+                {
+                    Role = UserRole.Banned,
+                }
+            }
+        )).Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.Unauthorized);
+    }
+
+    [TestCase(UserRole.Confirmed)]
+    [TestCase(UserRole.Registered)]
+    [TestCase(UserRole.Banned)]
+    public async Task BanUser_NonAdmin_Unauthorized(UserRole role)
```

Comment:
Forbidden, not Unauthorized (to match the HTTP rseponse code names). One instance below.